### ISSUE TITLE
Periodic jobs if worker timezone is different than the ruby system timezone

### ISSUE THREAD

**Issue by aliismayilov** [mperham/sidekiq#4796]
Ruby version: ruby 2.7.2p137 (2020-10-01 revision 5445e04352) [x86_64-darwin20]
activesupport: 6.1.1
Sidekiq / Pro / Enterprise version(s): 6.1.3 / 5.2.1 / 2.2.1

```ruby
require "sidekiq-ent"
require "active_support/all"

WORKER_TIMEZONE = "EST"

class PeriodicWorker
  include Sidekiq::Worker

  def perform
    puts "PeriodicWorker workin..."
  end
end

Sidekiq.configure_server do |config|
  config.periodic do |mgr|
    mgr.tz = ActiveSupport::TimeZone.new(WORKER_TIMEZONE)
    mgr.register("* * * * *", "PeriodicWorker")
  end
end

# run with
# TZ=UTC ./bin/sidekiq -r ./por.rb
```

Assuming that the system timezone is UTC:
* If `WORKER_TIMEZONE = "EST"`, then no worker is enqueued.
* If `WORKER_TIMEZONE = "UTC"`, then worker is enqueued as expected, once a minute.
* If `WORKER_TIMEZONE = "Kathmandu"`, then worker gets enqueued in every loop.

**Comment by mperham** [comment]
Confirmed. I'm not clear what's wrong but it looks like I'll be spending some time in byebug today.

**Comment by mperham** [comment]
I've released v2.2.2, how does it look to you?

**Comment by aliismayilov** [comment]
seems to be working. thanks!
